source/Calamari.Aws/Deployment/Conventions/ExecuteCloudFormationChangeSetConvention.cs

```diff
@@ -49,15 +50,22 @@ private async Task InstallAsync(RunningDeployment deployment)
 
             Guard.NotNull(stack, "The provided stack identifer or name may not be null");
             Guard.NotNull(changeSet, "The provided change set identifier or name may not be null");
-
-            var result = await ExecuteChangeset(clientFactory, stack, changeSet);
-
-            if (result.None())
+            
+            var response = await clientFactory.DescribeChangeSetAsync(stack, changeSet);
+            if (response.Changes.Count == 0)
             {
+                await clientFactory().DeleteChangeSetAsync(new DeleteChangeSetRequest
+                {
+                    ChangeSetName = changeSet.Value,
+                    StackName = stack.Value
+                });
+                
                 Log.Info("No changes changes are to be performed.");
```

Comment:
Hehe. "No changes changes..." :)

Reply:
Oh dear me. I am having one of those days...